tests/contrib/openai/test_openai_v1.py

```diff
@@ -1346,3 +1346,114 @@ async def handle_async_request(self, request: httpx.Request) -> httpx.Response:
         assert False, f"Unexpected exception: {e}"
 
     assert asyncio_timeout, "Expected asyncio.TimeoutError"
+
+
+@pytest.mark.skipif(
+    parse_version(openai_module.version.VERSION) < (1, 66), reason="Response options only available openai >= 1.66"
+)
+@pytest.mark.snapshot(token="tests.contrib.openai.test_openai.test_response_completion")
+def test_response_completion(openai, mock_tracer):
+    """Ensure llmobs records are emitted for response endpoints when configured."""
+    with get_openai_vcr(subdirectory_name="v1").use_cassette("response_create.yaml"):
+        model = "gpt-4.1"
+        input_messages = multi_message_input
+        client = openai.OpenAI()
+        client.responses.create(
+            model=model, input=input_messages, top_p=0.9, max_output_tokens=100, user="ddtrace-test"
+        )
+    span = mock_tracer.pop_traces()[0][0]
+    assert span.name == "openai.request"
+    assert span.resource == "createResponse"
+    assert span.get_tag("openai.request.model") == "gpt-4.1"
+
+
+@pytest.mark.skipif(
+    parse_version(openai_module.version.VERSION) < (1, 66), reason="Response options only available openai >= 1.66"
+)
+@pytest.mark.parametrize("api_key_in_env", [True, False])
+def test_response_completion_error(openai, mock_tracer, request_api_key):
+    """Assert errors when an invalid model is used."""
+    # import openai
+    with get_openai_vcr(subdirectory_name="v1").use_cassette("response_create_error.yaml"):
+        with pytest.raises(openai.BadRequestError):
+            # Create client directly without using the openai fixture
+            client = openai.OpenAI(api_key=request_api_key)
+            client.responses.create(
+                model="invalid-model",  # Using an invalid model to trigger error
+                input="Hello world",
+                user="ddtrace-test",
+            )
+    span = mock_tracer.pop_traces()[0][0]
+    assert span.name == "openai.request"
+    assert span.resource == "createResponse"
+    assert span.get_tag("openai.request.model") == "invalid-model"
+    assert span.error == 1
+    assert span.get_tag("error.type") == "openai.BadRequestError"
+    assert span.get_tag("error.message") is not None
+
+
+@pytest.mark.skipif(
+    parse_version(openai_module.version.VERSION) < (1, 66), reason="Response options only available openai >= 1.66"
+)
+@pytest.mark.parametrize("api_key_in_env", [True, False])
+async def test_aresponse_completion(openai, mock_tracer, request_api_key, openai_vcr):
+    """Assert spans are created with async client."""
+    with openai_vcr.use_cassette("test_aresponse_completion.yaml"):
```

Comment:
Can we reuse the `response.yaml` cassette for both sync/async tests (in the context of minimizing files to maintain)?